src/components/colorPicker/index.js

```diff
@@ -118,6 +121,12 @@ export default class ColorPicker extends PureBaseComponent {
     }
   }
 
+  showMigrationMessage() {
+    console.warn(`ColorPicker component will soon become uncontrolled component.` +
+    ` Thus 'initialColor' prop will become INITIAL, meaning, no updates to this prop will take effect` +
+    ` unless passing 'key' prop.`);
+  }
```

Comment:
I noticed this warnings pops every time we select a color in the palette (in the example screen) 
I think it might be a problem not supporting the current behavior. 

In One App we have this scenario when the user can select a color from a palette and it affect the ColorPicker as well..

Reply:
Yes, the warning is in componentWillRecieveProps() to alert when using 'initialColor' prop to update the picker. 
Regarding the functionality, it will be the same as long as the users will ADD the 'key' prop with the state color. This way we keep the component uncontrolled but let the user update using state getting a new instance of the component when the 'initialColor' (and 'key') changes. An example is on ColorPickerScreen line #83.

Reply:
Ok, I went over the component implementation again.. 
The component is actually kind of controlled. which is good. 
We need to separate the two parts of the component... 

One part is the color picker field (the add button with the colors palette) - This part is controlled cause it depends on the user's `value` prop - which is good!

The second part is the color picker dialog (the "editor") which clearly needs to have an internal state. 
That means that we can separate the "editor" part to an internal component and always initialize it when opened, with the current `value` prop. 
The fact that the "editor" has internal state doesn't change the fact the ColorPicker is controlled. 
Eventually, we determine the "current" value according to what the user pass in props, and that's what matter. 

Anyway, we don't need both `value` and `initialValue` we can have only `value`. 

Let me know if what I wrote isn't clear, we can sync on this on Sunday.

Reply:
Love it. Done.